### PR TITLE
Dependency pinning

### PR DESCRIPTION
**Pull request by hhsecond**
## What does this PR do?

In the light of recent starsessions version conflict that disrupted the lightning cloud service, let's pin all the dependencies to a safe range without keeping it wide open. This PR pins the croniter package and relaxes redis's pinning a bit

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: requirements/app/base.txt [hunk 1]
```diff
@@ -4,6 +4,6 @@ deepdiff>=5.7.0, <=5.8.1
 starsessions>=1.2.1, <2.0 # strict
 fsspec>=2022.01.0, <=2022.7.1
 s3fs>=2022.1.0, <=2022.7.1
-croniter # for now until we found something more robust.
+croniter~=1.3.0 # for now until we found something more robust.
```
[the remark this answers is not in the document]
Comment (Borda) [reply requirements/app/base.txt:7]:
```suggestion
croniter==1.3.* # for now until we found something more robust.
```

**Review by tchaton — APPROVED** [review]
LGTM !

File: requirements/app/base.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Borda) [reply requirements/app/base.txt:7]: Ok, personally I would go with `*`, but I am fine with this one too, let's just check that the requirements parsing are correct...
one more note, could you also check that dependable takes it?
```suggestion
croniter>=1.3.0, <1.4.0 # for now until we found something more robust.
```

**Review by rohitgr7 — APPROVED** [review]

File: requirements/app/cloud.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
-redis>=4.0.0, <=4.2.4
+redis>=4.0.0, <5.0.0
```
Comment (Borda) [inline requirements/app/cloud.txt:1]: Is this meant to be strict?
still, this won't prevent accidental failures in our CI for e.g. 4.3.0

**Review by Borda — CHANGES_REQUESTED** [review]
see: https://github.com/Lightning-AI/lightning/pull/14463#discussion_r959464454

File: requirements/app/base.txt [hunk 3]
```diff
@@ -4,6 +4,6 @@ deepdiff>=5.7.0, <=5.8.1
 starsessions>=1.2.1, <2.0 # strict
 fsspec>=2022.01.0, <=2022.7.1
 s3fs>=2022.1.0, <=2022.7.1
-croniter # for now until we found something more robust.
+croniter==1.3.* # for now until we found something more robust.
```
Comment (Borda) [inline requirements/app/base.txt:7]:
```suggestion
croniter>=1.3.0, <1.4.0  # strict; TODO: for now until we find something more robust.
```

**Review by Borda — APPROVED** [review]

File: src/lightning_app/CHANGELOG.md [hunk 4]
```diff
@@ -31,10 +31,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Changed
 
 - Default values and parameter names for Lightning AI BYOC cluster management ([#14132](https://github.com/Lightning-AI/lightning/pull/14132))
-
-### Changed
-
 - Run the flow only if the state has changed from the previous execution ([#14076](https://github.com/Lightning-AI/lightning/pull/14076))
+- Pinning croniter to 1.3.x ([#14463](https://github.com/Lightning-AI/lightning/pull/14463))
```
Comment (Borda) [inline src/lightning_app/CHANGELOG.md:35]:
```suggestion
```
I think this is not needed, already handled by python depenecies

**Pull request merged**